sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/NullPointerDereference.CSharp9.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Text;
+
+object topLevel = null;
+topLevel.ToString();   // FN
+
+topLevel = "Value";
+topLevel.ToString();
+
+void TopLevelLocalFunction()
+{
+    object local = null;
+    local.ToString();
+
+    local = "Value";
+    local.ToString();
+}
+
+public class Sample
+{
+    private string field;
+
+    public void TargetTypedNew()
+    {
+        StringBuilder sb = null;
+        sb.ToString(); // FN, can't build CFG for this method
+
+        sb = new(42);
+        sb.Append("Value");
+    }
+
+    public void PatternMatching(object arg)
+    {
+
+        if (arg is string)
+        {
+            arg.ToString();     // Compliant
+        }
+
+        if (arg is int and > 0 and > 1)
+        {
+            arg.ToString();     // Compliant
+        }
+
+        if (arg is int or bool or long)
+        {
+            arg.ToString();     // Compliant
+        }
+
+        if (arg is null)
```

Comment:
I would add a test case also with a noncompliant example with `not`. Something similar with:
```
if (o is not not null)
{
    o.ToString();
}
```